main/src/main/java/org/mobilitydata/gtfsvalidator/validator/UrlConsistencyValidator.java

```diff
@@ -70,72 +59,47 @@
 
   @Override
   public void validate(NoticeContainer noticeContainer) {
-    Map<String, GtfsAgency> agencyByUrlMap = agenciesByUrlMap(agencyTable);
+    ListMultimap<String, GtfsAgency> agencyByUrlMap = agenciesByUrlMap(agencyTable);
+    ListMultimap<String, GtfsRoute> routesByUrlMap = routesByUrlMap(routeTable);
     for (GtfsRoute route : routeTable.getEntities()) {
       if (!route.hasRouteUrl()) {
-        return;
+        continue;
       }
-      GtfsAgency agency = agencyByUrlMap.get(route.routeUrl().toLowerCase());
-      if (agency != null) {
-        noticeContainer.addValidationNotice(
-            new SameRouteAndAgencyUrlNotice(
-                route.csvRowNumber(),
-                route.routeId(),
-                agency.agencyName(),
-                route.routeUrl(),
-                agency.csvRowNumber()));
+      List<GtfsAgency> agencies = agencyByUrlMap.get(route.routeUrl().toLowerCase());
+      for (GtfsAgency agency : agencies) {
+        noticeContainer.addValidationNotice(new SameRouteAndAgencyUrlNotice(route, agency));
       }
     }
     for (GtfsStop stop : stopTable.getEntities()) {
       if (!stop.hasStopUrl()) {
-        return;
+        continue;
       }
-      GtfsAgency agency = agencyByUrlMap.get(stop.stopUrl().toLowerCase());
-      if (agency != null) {
-        noticeContainer.addValidationNotice(
-            new SameStopAndAgencyUrlNotice(
-                stop.csvRowNumber(),
-                stop.stopId(),
-                agency.agencyName(),
-                stop.stopUrl(),
-                agency.csvRowNumber()));
+      List<GtfsAgency> agencies = agencyByUrlMap.get(stop.stopUrl().toLowerCase());
+      for (GtfsAgency agency : agencies) {
+        noticeContainer.addValidationNotice(new SameStopAndAgencyUrlNotice(stop, agency));
+      }
+      List<GtfsRoute> routes = routesByUrlMap.get(stop.stopUrl().toLowerCase());
+      for (GtfsRoute route : routes) {
+        noticeContainer.addValidationNotice(new SameStopAndRouteUrlNotice(stop, route));
       }
     }
-    Map<String, GtfsRoute> routesByUrlMap = routesByUrlMap(routeTable);
-    Map<String, GtfsStop> stopsByUrlMap = stopsByUrlMap(stopTable);
-
-    Maps.filterEntries(
-            stopsByUrlMap, entry -> routesByUrlMap.get(entry.getValue().stopUrl()) != null)
-        .values()
-        .forEach(
-            stopWithDuplicateUrl -> {
-              noticeContainer.addValidationNotice(
-                  new SameStopAndRouteUrlNotice(
-                      stopWithDuplicateUrl.csvRowNumber(),
-                      stopWithDuplicateUrl.stopId(),
-                      stopWithDuplicateUrl.stopUrl(),
-                      routesByUrlMap.get(stopWithDuplicateUrl.stopUrl()).routeId(),
-                      routesByUrlMap.get(stopWithDuplicateUrl.stopUrl()).csvRowNumber()));
-            });
   }
 
   /**
    * Maps {@code GtfsAgency}s by there URLs if provided.
```

Comment:
Typo: by *their* URLs. There and their are homophones.

Also elsewhere.

Reply:
Thanks :)